lib/src/annotation.dart

```diff
@@ -51,33 +49,35 @@ dynamic instantiateAnnotation(ElementAnnotation annotation) {
       "${valueDeclaration.runtimeType}.");
 }
 
-dynamic _getValue(DartObject object, TypeProvider typeProvider) {
-  if (object.isNull) {
+dynamic _getValue(DartObject object) {
+  var reader = new ConstantReader(object);
+
+  if (reader.isNull) {
```

Comment:
Sadly no, we need to talk about that.

Type, List, and Map are weird – you get DartType.

I think we should update `anyValue` to exclude those and explain why.